### PR TITLE
[KYUUBI #4331] [KYUUBI #4431] Lineage supports `Union` and `MergeInto`

### LINKED ISSUE #4331: [FEATURE] [Lineage] `MergInto` statement support the iceberg table
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature
image

### Motivation

_No response_

### Describe the solution

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### LINKED ISSUE #4431: [FEATURE][Lineage] Support `FROM INSERT xxx INSERT xxx` statement
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

Test case:

```
test("table FROM INSERT") {
  withTable("t1", "t2") { _ =>
    spark.sql("CREATE TABLE t1 (a string, b string) USING hive")
    spark.sql("CREATE TABLE t2 (a string, b string) USING hive")
    spark.sql("CREATE TABLE t3 (a string, b string) USING hive")
    val ret0 = exectractLineage("from (select a,b from t1)" +
        " insert overwrite table t2 select a,b where a=1" +
        " insert overwrite table t3 select a,b where b=1")
  assert(ret0 == Lineage(
    List(),
    List(),
    List()))
  }
}
```

### Motivation

_No response_

### Describe the solution

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by iodone**
close #4331 
close #4431 

### _Why are the changes needed?_

Because `optimizedPlan` has been replaced by `analyzedPlan`, iceberg tables do not need to be additionally processed when new logical plans are generated during some optimization stages. They can be treated as regular tables.

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by iodone** [comment]
cc @ulysses-you

File: extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala [hunk 1]
```diff
@@ -376,14 +376,20 @@ trait LineageParser {
...
       case p: Union =>
-        // merge all children in to one derivedColumns
-        val childrenUnion =
-          p.children.map(extractColumnsLineage(_, ListMap[Attribute, AttributeSet]())).map(
-            _.values).reduce {
-            (left, right) =>
-              left.zip(right).map(attr => attr._1 ++ attr._2)
-          }
-        val childrenColumnsLineage = ListMap(p.output.zip(childrenUnion): _*)
+        val childrenColumnsLineage = if (p.output.isEmpty) {
```
Comment (ulysses-you) [inline extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala:379]: can we cdd some comments about, this is for multi-insert clause

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master/1.7

**Comment by pan3793** [comment]
this patch cause branch-1.7 CI failed, could you take a look? @ulysses-you @iodone

**Comment by iodone** [comment]
> this patch cause branch-1.7 CI failed, could you take a look? @ulysses-you @iodone
I will follow up.

**Pull request closed without merging**